apps/site/components/Common/Search/index.tsx

```diff
@@ -1,61 +1,126 @@
 'use client';
 
-import { MagnifyingGlassIcon } from '@heroicons/react/24/outline';
-import classNames from 'classnames';
-import { useTranslations } from 'next-intl';
-import { useState, type FC } from 'react';
+import { OramaSearchBox, OramaSearchButton } from '@orama/react-components';
+import { useRouter } from 'next/navigation';
+import { useTranslations, useLocale } from 'next-intl';
+import { useTheme } from 'next-themes';
+import { type FC } from 'react';
 
-import { WithSearchBox } from '@/components/Common/Search/States/WithSearchBox';
-import { useDetectOS } from '@/hooks';
-import { useKeyboardCommands } from '@/hooks/react-client';
+import {
+  ORAMA_CLOUD_ENDPOINT,
+  ORAMA_CLOUD_API_KEY,
+  DEFAULT_ORAMA_QUERY_PARAMS,
+  DEFAULT_ORAMA_SUGGESTIONS,
+  BASE_URL,
+} from '@/next.constants.mjs';
 
-import styles from './index.module.css';
+type ResultMapDescription = {
+  path: string;
+  pageSectionTitle: string;
+};
+
+type ResultMapPath = { path: string; siteSection: string };
+
+import { themeConfig } from './utils';
+
+const uppercaseFirst = (word: string) =>
+  word.charAt(0).toUpperCase() + word.slice(1);
 
-export const SearchButton: FC = () => {
-  const [isOpen, setIsOpen] = useState(false);
+const getFormattedPath = (path: string, title: string) =>
+  `${path
+    .replace(/#.+$/, '')
+    .split('/')
+    .map(element => element.replaceAll('-', ' '))
+    .map(element => uppercaseFirst(element))
+    .filter(Boolean)
+    .join(' > ')} — ${title}`;
+
+const SearchButton: FC = () => {
+  const { resolvedTheme } = useTheme();
   const t = useTranslations();
-  const openSearchBox = () => setIsOpen(true);
-  const closeSearchBox = () => setIsOpen(false);
-
-  useKeyboardCommands(cmd => {
-    switch (cmd) {
-      case 'cmd-k':
-        openSearchBox();
-        break;
-      case 'escape':
-        closeSearchBox();
-        break;
-      default:
-    }
-  });
-
-  const { os } = useDetectOS();
-
-  const osCommandKey = os === 'MAC' ? '⌘' : 'Ctrl';
-  const isOSLoading = os === 'LOADING';
+  const locale = useLocale();
+  const colorScheme = resolvedTheme as 'light' | 'dark';
+  const router = useRouter();
```

Comment:
You're using the wrong router component. You should use the `useRouter` from `@/navigation.mjs`, this is causing the Storybooks to fail.

Reply:
Hi! I did not use it at first because it always concatenates the locale, and we do not want this behavior for /docs.
I'm now using it with locale set to undefined, was we already display the URL with the right locale on the SearchBox (using resultMap).